packages/swapper/src/swappers/thorchain/utils/txFeeHelpers/btcTxFees/getBtcTxFees.ts

```diff
@@ -1,49 +1,51 @@
-import { bitcoin, ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
+import { ChainId } from '@shapeshiftoss/caip'
+import { ChainAdapter } from '@shapeshiftoss/chain-adapters'
 import { KnownChainIds } from '@shapeshiftoss/types'
 
 import { QuoteFeeData, SwapError, SwapErrorTypes } from '../../../../../api'
 import { bn } from '../../../../utils/bignumber'
+import { ThorChainUtxoChainIds } from '../../../types'
 
 export const getBtcTxFees = async ({
   opReturnData,
   vault,
   sellAmount,
-  adapterManager,
+  sellAdapter,
   pubkey,
   tradeFee
 }: {
   opReturnData: string
   vault: string
   sellAmount: string
-  adapterManager: ChainAdapterManager
+  sellAdapter: ChainAdapter<ChainId>
   pubkey: string
   tradeFee: string
 }): Promise<QuoteFeeData<KnownChainIds.BitcoinMainnet>> => {
   try {
-    const adapter = adapterManager.get('bip122:000000000019d6689c085ae165831e93') as
-      | bitcoin.ChainAdapter
-      | undefined
-    if (!adapter)
-      throw new SwapError(
-        `[getBtcTxFees] - No chain adapter found for bip122:000000000019d6689c085ae165831e93.`,
-        {
-          code: SwapErrorTypes.UNSUPPORTED_CHAIN,
-          details: { chainId: 'bip122:000000000019d6689c085ae165831e93' }
-        }
-      )
-    const feeDataOptions = await adapter.getFeeData({
+    const feeDataOptions = await (sellAdapter as ChainAdapter<ThorChainUtxoChainIds>).getFeeData({
       to: vault,
       value: sellAmount,
       chainSpecific: { pubkey, opReturnData }
     })
 
     const feeData = feeDataOptions['fast']
 
+    // BCH specific hack:
+    // For some reason when sats per byte comes back as 1 (which is very common for bch)
+    // broadcast will fail because it thinks the intrinsic fee is too low
+    // it feels like possibly an off by-a-few-bytes bug with how we are using coinselect with opReturnData
+    // Bumping BCH fees here resolves this for now until we have time to find a better solution
+    const isFromBch = sellAdapter.getChainId() === KnownChainIds.BitcoinCashMainnet
```

Comment:
- Make this a function, which will allow it to be a type guard?

```suggestion
    const isBchAdapter = (
      adapter: UtxoBaseAdapter<ThorChainUtxoChainIds>
    ): adapter is UtxoBaseAdapter<KnownChainIds.BitcoinCashMainnet> =>
      adapter.getChainId() === KnownChainIds.BitcoinCashMainnet

    const isBchSellAsset = isBchAdapter(sellAdapter)
```

- Note the separate type guard definition and assignment in `isBchSellAsset` which IMO separates things better, and allows us to possibly reuse this type guard

Reply:
Its only used in this one place. Feels like an unnecessary abstraction

Reply:
I won't block the PR for it, but conceptually, `isXYZ` as an expression is just a suboptimal type guard IMO.

This isn't so much about us needing an abstraction (although we *might* need it in the future which I agree with you sounds like early abstraction) than it is about making sure we follow clean patterns that allow the codebase to be maintainable, and the next contributor not copying and pasting it somewhere they would need type narrowing, then getting rugged.

Reply:
I disagree.  Im not really a fan of pre-abstraction lbefore the pattern is going to be used in one place.

In my experience I have found its easy to get into trouble by assuming what we will need before we actually need it. It leads to ver engineering and too much complexity